### ISSUE TITLE
RStudio startup error with v1.4.52

### ISSUE THREAD

**Issue by pat-s** [rstudio/rstudio#6479]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.52
    OS Version      : macOS 10.15.3
    R Version       : 3.63

### Steps to reproduce the problem

Start RStudio.

### Describe the problem in detail

R vanilla start (no .Rprofile files). Current RStudio Preview (1.3.911) works normally.

PROCESS OUTPUT
The R session process exited with code 6.

ERRORS
```
dyld: Library not loaded: /opt/rstudio-tools/soci/build/lib/libsoci_core.4.0.dylib
Referenced from: /Applications/RStudio.app/Contents/MacOS/rsession
Reason: image not found

```
### Describe the behavior you expected

Normal start.

**Comment by jmcphers** [comment]
@kfeinauer The way we deal with this for OpenSSL is to bundle the dylibs and then rewrite the references to be relative to the executable path. 

https://github.com/rstudio/rstudio/blob/28a5e3ff51462e836e4cd002507853ec92de87ee/src/cpp/session/CMakeLists.txt#L629-L647

**Comment by kfeinauer** [comment]
@jmcphers I saw that - am attempting a similar approach now.

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.4.724, R Version - 3.6.3, Mac OS 10.15.6 - Desktop 

Was able to reproduce the reported error state on the 1.4.52 build, confirmed it no longer occurs on the latest [Daily build](https://dailies.rstudio.com/).
